ignite/metrics/epoch_metric.py

```diff
@@ -139,6 +141,11 @@ def compute(self) -> float:
         if idist.get_rank() == 0:
             # Run compute_fn on zero rank only
             result = self.compute_fn(_prediction_tensor, _target_tensor)
+            if result is not None:
+                try:
+                    apply_to_type(result, (torch.Tensor, float), partial(idist.broadcast, src=0))
```

Comment:
This is wrong and this makes machine to hang

Reply:
The point was to raise a more comprehensive error message.
The idea is that we now add a feature to handle other types than only `torch.Tensor`: tuple, list or mapping of numbers, tensors etc. We also have to update the docstring for EpochMetric saying about supported outputs of compute_fn.
Well, I have to think a bit more about how to do that properly...

Reply:
I tried the solution you proposed 
```
if idist.get_rank() == 0:
    # Run compute_fn on zero rank only
    result = self.compute_fn(_prediction_tensor, _target_tensor)
    if result is not None:
        apply_to_type(result, (torch.Tensor, float, str), partial(idist.broadcast, src=0))
```
but it didn't pass the `test_epoc_metric` because EpocMetric inputs int sometimes so i changed it to 
```
if idist.get_rank() == 0:
    # Run compute_fn on zero rank only
    result = self.compute_fn(_prediction_tensor, _target_tensor)
    if result is not None:
        apply_to_type(result, (torch.Tensor, float, int, str), partial(idist.broadcast, src=0))
```
but that's obviously doesn't work with mypy, the second arg must be tuple, so i did it the way you see

Reply:
```
if idist.get_rank() == 0:
    # Run compute_fn on zero rank only
    result = self.compute_fn(_prediction_tensor, _target_tensor)
    if result is not None:
        apply_to_type(result, (torch.Tensor, float, str), partial(idist.broadcast, src=0))
```
Let me explain why this is wrong:
- `apply_to_type` returns an output and is not inplace as you use it
- we can not call `idist.broadcast` for a single proc only (if idist.get_rank() == 0)